py/server/deephaven/experimental/iceberg.py

```diff
@@ -270,6 +271,90 @@ def from_current_snapshot(cls) -> 'SchemaProvider':
         return cls(_JSchemaProvider.fromCurrentSnapshot())
 
 
+class SortOrderProvider(JObjectWrapper):
+    """
+    :class:`.SortOrderProvider` is used for providing SortOrder to be used for sorting new data while writing to an
+    iceberg table using this writer. Users can specify multiple ways to do so, for example, by sort ID, table default,
+    etc.
+    """
+
+    j_object_type = _JSortOrderProvider
+
+    def __init__(self, _j_object: jpy.JType):
+        """
+        Initializes the :class:`.SortOrderProvider` object.
+
+        Args:
+            _j_object (SortOrderProvider): the Java :class:`.SortOrderProvider` object.
+        """
+        self._j_object = _j_object
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+    @classmethod
+    def unsorted(cls) -> 'SortOrderProvider':
+        """
+        Used to disable sorting while writing new data to the iceberg table.
+
+        Returns:
+            the SortOrderProvider object.
+        """
+        return cls(_JSortOrderProvider.unsorted())
+
+    @classmethod
+    def use_table_default(cls) -> 'SortOrderProvider':
+        """
+        Use the default sort order of the table while writing new data. If no sort order is set on the table, no sorting
+        will be done.
+
+        Returns:
+            the :class:`.SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.useTableDefault())
+
+    @classmethod
+    def from_sort_id(cls, sort_order_id: int) -> 'SortOrderProvider':
+        """
+        Use the sort order with the given ID to sort new data while writing to the iceberg table.
+
+        Args:
+            sort_order_id (int): the id of the sort order to use.
+
+        Returns:
+            the :class:`.SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.fromSortId(sort_order_id))
+
+    def with_id(self, sort_order_id: int) -> 'SortOrderProvider':
+        """
+        Returns a sort order provider that delegates to this provider for computing the columns to sort on, but writes a
+        different sort order ID to the iceberg table.
+        For example, this provider might return fields {A, B, C} to sort on, but the ID written to iceberg corresponds
+        to sort order with fields {A, B}.
+
+        Args:
+            sort_order_id (int): the sort order ID to write to the iceberg table.
```

Comment:
tables have sort order IDs stored with them?  This is the kind of stuff that isn't documented well enough.  No part of this documentation would have suggested such a thing to me.  There is a lot of assumed knowledge that needs to be explicit in the docs for the new code.